### PR TITLE
PE-2771: read app tags iteratively instead of just the first match

### PR DESCRIPTION
**Pull request by matibat**
Changes
- Reads the App Tags iteratively. Previously only the first match would be checked.

Extra :-)
- Explicitly sets the missing type of some methods

### CODE REVIEW

File: src/common.ts [hunk 1]
```diff
@@ -197,11 +197,18 @@ export const ardriveOracle = new ArDriveContractOracle([
...
+
 	const isV2Tx = !bundledIn;
-	if (appName === WEB_APP_NAME && appVersion && !isSemanticVersionGreaterThan(appVersion, '1.14.1') && isV2Tx) {
+	const isWebApp = appNameTags.some((appNameTag) => appNameTag.value === WEB_APP_NAME);
+	const isAppVersionGreaterThan1_14_1 = appVersionTags.some((appVersionTag) =>
+		// FIXME: this way of doing it cannot distinguish between multiple apps' version
+		// might result in false positives
+		isSemanticVersionGreaterThan(appVersionTag.value, '1.14.1')
+	);
+	if (isWebApp && !isAppVersionGreaterThan1_14_1 && isV2Tx) {
```
Comment (matibat) [inline src/common.ts]: Should we have an array of ArDrive versions somewhere just to be more precise? Hard to track?

Comment (karlprieb) [reply src/common.ts]: Hard to track and even with that there's a possibility of other apps using App-Version to have the same versions that we want.

Comment (karlprieb) [reply src/common.ts]: Just provide a comment here saying that until SmartWeaveAction reaches v1.14.1 we're good

Comment (matibat) [reply src/common.ts]: There it is - a732b61245cd55f6bc21925a1c7adb0e2d0f7c09

**Review by karlprieb — APPROVED** [review]

**Pull request merged**